lib/ex_machina/ecto_strategy.ex

```diff
@@ -3,10 +3,20 @@ defmodule ExMachina.EctoStrategy do
 
   use ExMachina.Strategy, function_name: :insert
 
-  def handle_insert(%{__meta__: %{__struct__: Ecto.Schema.Metadata}} = record, %{repo: repo}) do
-    repo.insert! record
+  def handle_insert(%{__meta__: %{state: :loaded}} = record, _) do
+    raise ArgumentError,
+    "you called `insert` on a record that has already been inserted.
+     You can only call `insert` on records that haven't been inserted yet.
```

Comment:
Now that I'm reading this again it seems a bit redundant. Maybe the second sentence should be "Make sure that you have not accidentally called `insert` twice." What do you think?